### PR TITLE
Add biox v1.0.0 - A universal biological sequence compressor

### PR DESCRIPTION
**Pull request by TianMayCry9**
Describe your pull request here

# Description
This is the first official release of BioX, a specialized compression tool for biological sequence data. 
﻿
## Key Features
* High-performance biological sequence compression tool
* Supports FASTA and FASTQ formats
* Handles DNA, RNA, and protein sequences
* Project URL: https://github.com/TianMayCry9/BioX
﻿
## Technical Highlights
* Fully tested on Windows and Linux platforms
* Validated with genomic data from 100+ species and diverse biological sequence types
* Offers rapid processing at low compression levels while maintaining effective compression ratios
* Achieves industry-leading compression strength at high compression levels
﻿
## Testing
* Compression/decompression tests on various biological sequence formats
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by TianMayCry9** [comment]
@BiocondaBot please add label

File: recipes/biox/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  run_exports:
+    - {{ pin_subpackage('biox', max_pin="x") }}
+
+requirements:
+  build:
```
Comment (martin-g) [inline recipes/biox/meta.yaml:20]: There is no need of build requirements for Python

File: recipes/biox/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/TianMayCry9/BioX/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: ecbaf5d74f93361fdfa1798fc34edc27b18b131b2943f66da1347856444fb9f1
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
```
Comment (martin-g) [inline recipes/biox/meta.yaml:15]: Please see the other Python recipes. They use a bit longer command. 
Sorry for not being more specific but I am on my mobile

File: recipes/biox/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+test:
+  commands:
+    - biox -h
+    - printf ">test_seq\nACGTCGCGCGCGCG\n" > test.fa
+    - biox -c -t dna test.fa
+
+
+about:
+  home: https://github.com/TianMayCry9/BioX
+  license: GPL-3.0
```
Comment (martin-g) [inline recipes/biox/meta.yaml:45]:
```suggestion
  license: GPL-3.0-only
```
Spdx id

File: recipes/biox/meta.yaml (hunk 2 above)
Comment (TianMayCry9) [reply recipes/biox/meta.yaml]: Thank you for your feedback! I have made the changes based on your suggestions. Please review them and let me know if they meet your expectations. Feel free to share any additional questions or further suggestions!

**Review by mbhall88 — APPROVED** [review]

**Comment by TianMayCry9** [comment]
@mbhall88 Thank you for reviewing and merging my PR. I would like to join the Bioconda organization to maintain this tool and contribute more to the community. Could you guide me on the next steps for becoming a member? Thank you!

**Comment by mbhall88** [comment]
Because you are listed as the recipe maintainer you will be notified whenever a version bump is triggered for this package.

As for becoming part of the bioconda org, that is a question better asked of @bioconda/core maybe?

**Pull request merged**